src/Tasks/Microsoft.Common.CurrentVersion.targets

```
@@ -611,6 +611,11 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <ContinueOnError Condition="'$(ContinueOnError)' == ''">false</ContinueOnError>
   </PropertyGroup>
 
+  <!-- Users familiar with how some other repos work try to use NoWarn with MSBuild in place of MSBuildWarningsAsMessages. -->
+  <PropertyGroup Condition="'$(MSBuildWarningsAsMessages)'=='' And $(MSBuildChangeWaveVersion) != '' And $([MSBuild]::VersionGreaterThanOrEquals('$(MSBuildChangeWaveVersion)', '16.8'))">
```

```suggestion
  <PropertyGroup Condition="'$(MSBuildWarningsAsMessages)'=='' And $([MSBuild]::VersionGreaterThanOrEquals('$(MSBuildChangeWaveVersion)', '16.8'))">
```

`MSBuildChangeWaveVersion` is guaranteed to have a proper value, so this check isn't necessary.

---

That's true whether or not the user specifies a value? And does that also get pulled in for random tests?

---

Yes, each project will have `MSBuildChangeWaveVersion` set as a reserved property. If unset, it currently defaults to `999.999`.

> And does that also get pulled in for random tests?
I'm not sure what you mean?

---

I added that because unit tests failed without it. Unit tests often don't have a proper project/run a proper build, and I haven't checked whether these particular tests do or don't. Is that a problem?

---

This is actually a great point and something that'll have to be added to the docs. I had to make sure to build using a `ProjectCollection` so that the reserved properties would be set at all. Tests that need to build should be done similar to this:

```c#
                string projectFile = @"
                    <Project>
                        <Target Name='HelloWorld' Condition=""'$(MSBUILDCHANGEWAVEVERSION)' == '999.999' and $([MSBuild]::VersionLessThan('" + featureWave + @"', '$(MSBUILDCHANGEWAVEVERSION)'))"">
                            <Message Text='Hello World!'/>
                        </Target>
                    </Project>";

                TransientTestFile file = env.CreateFile("proj.csproj", projectFile);

                ProjectCollection collection = new ProjectCollection();
                MockLogger log = new MockLogger();
                collection.RegisterLogger(log);

                collection.LoadProject(file.Path).Build().ShouldBeTrue();
```

Are there a significant amount of tests that fail here?